Tests/SwiftDiagnosticsTest/FixItTests.swift

```diff
@@ -0,0 +1,37 @@
+import SwiftParser
+import SwiftParserDiagnostics
+import SwiftSyntax
+import XCTest
+import _SwiftSyntaxTestSupport
+
+final class FixItTests: XCTestCase {
+  func testEditsForFixIt() {
+    let testCases: [[(range: Range<Int>, replacement: String)]] = [
+        [(9..<15, "Multiident "), (15..<21, "")],
+        [(9..<15, "MultiIdent "), (15..<21, "")]
+      ]
+
+    let markedSource = "protocol Multi 1️⃣ident {}"
+    let (_, source) = extractMarkers(markedSource)
+    let tree = Parser.parse(source: source)
+    let diags = ParseDiagnosticsGenerator.diagnostics(for: tree)
+    XCTAssertEqual(diags.count, 1)
+    let diag = diags.first!
+    XCTAssertEqual(diag.fixIts.count, 2)
+
+    for i in 0..<diag.fixIts.count {
+      let fixIt = diag.fixIts[i]
+      let changes = fixIt.changes
+      let edits = fixIt.edits
+      XCTAssertNotEqual(changes.count, edits.count)
+      let expctedEdits = testCases[i]
```

Comment:
This will crash if we end up having more than 2 Fix-Its, which makes it harder to debug the tests. Since both changes are pretty similar, I think it would be OK to just test the changes of the first Fix-It here. I can’t imagine any scenario where testing the second would catch an issue that we didn’t find with the first Fix-It.

Reply:
true, let me update the code to avoid the crash if test is failed.